src/s390x/zipl.rs

```diff
@@ -29,22 +39,159 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
-/// Runs `zipl` based on Ignition and BLS configuration in `boot`.
-pub fn zipl<P: AsRef<Path>>(boot: P) -> Result<()> {
+fn secure_execution_is_enabled() -> Result<bool> {
+    let sysfs_flag = "/sys/firmware/uv/prot_virt_guest";
+    match File::open(sysfs_flag) {
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(false),
+        Err(e) => Err(e).with_context(|| format!("reading {}", sysfs_flag)),
+        Ok(mut f) => {
+            let mut buffer = String::new();
+            f.read_to_string(&mut buffer)?;
+            Ok(buffer.trim() == "1")
+        }
+    }
+}
+
+fn find_files<P: AsRef<Path> + ?Sized>(
+    config_path: &P,
```

Comment:
Also should be `P` instead of `&P`. That should allow you to drop the `?Sized` marker.